src/js/actions/transform.js

```diff
@@ -1241,7 +1250,71 @@ define(function (require, exports) {
         reads: [locks.JS_APP, locks.JS_DOC],
         writes: [],
         transfers: ["history.newHistoryStateRogueSafe", "layers.resetBounds", "layers.resetSelection",
-            "layers.resetIndex"]
+            "layers.resetIndex"],
+        post: [layerActions._verifySelectedBounds]
+    };
+
+    /**
+     * Handle non-artboard layer transform events.
+     *
+     * @param {object} event
+     * @return {Promise}
+     */
+    var handleTransformLayer = function (event) {
+        this.dispatch(events.panel.TOGGLE_OVERLAYS, { enabled: true });
+
+        var appStore = this.flux.store("application"),
+            currentDoc = appStore.getCurrentDocument();
+
+        if (!currentDoc) {
+            return Promise.resolve();
+        }
+
+        // newDuplicateSheets move events should be processed immediately, not debounced
+        if (event.newDuplicateSheets) {
```

Comment:
This used to be debounced, but the debouncing was broken in the sense that it was functionally identical to just delaying the code from executing by 200ms. I couldn't think of any reason why that delay was necessary either, so I just removed it. It's entirely possible that this breaks something, but at least it's less mysterious...